streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -113,7 +113,7 @@ public StreamsMetadata getLocalMetadata() {
             return allMetadata;
         }
 
-        final Collection<String> sourceTopics = topologyMetadata.sourceTopicsForStore(storeName);
+        final Collection<String> sourceTopics = topologyMetadata.sourceTopicsForStore(storeName, null);
```

Comment:
Before this change, `StreamsMetadataState#getAllMetadataForStore(string)` returns the metadata for the store names across all topologies (it's possible that multiple topologies have stores with the same name), whereas now it is only returning for the `UNNAMED_TOPOLOGY`, is this change intentional?

Reply:
Yes, the reason is that a single store name is not sufficient to identify the actual store with named topologies, since a name may be reused for different stores in different topologies. So a user must use the overload that also accepts a `topologyName` parameter.

I will add a check that throws an informative exception if this method is called when named topologies are used, to be safe.